jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/core/PropertiesTest.java

```diff
@@ -31,35 +33,67 @@
 
     @Test
-    void testGetSettingWithCorrectPrefix() {
+    @SneakyThrows
+    public void testQuerySetting() {
+        val settings = Maps.immutableEntry("querySetting.date_style", "YMD");
+
+        assertWithStatement(
+                statement -> {
+                    val result = statement.executeQuery("SHOW date_style");
+                    result.next();
+                    assertThat(result.getString(1)).isEqualTo("ISO, YMD");
+                },
+                settings);
+    }
+
+    @Test
+    void testGetSettingWithCorrectPrefix() throws DataCloudJDBCException {
         Map<String, String> expected = ImmutableMap.of("lc_time", "en_US");
         Properties properties = new Properties();
         properties.setProperty(HYPER_SETTING + "lc_time", "en_US");
         properties.setProperty("username", "alice");
-        ConnectionQuerySettings connectionQuerySettings = ConnectionQuerySettings.of(properties);
-        assertThat(connectionQuerySettings.getSettings()).containsExactlyInAnyOrderEntriesOf(expected);
+        ConnectionProperties connectionProperties = ConnectionProperties.of(properties);
+        assertThat(connectionProperties.getStatementProperties().getQuerySettings())
+                .containsExactlyInAnyOrderEntriesOf(expected);
     }
 
     @Test
-    void testGetSettingReturnEmptyResultSet() {
+    void testGetSettingReturnEmptyResultSet() throws DataCloudJDBCException {
         Map<String, String> expected = ImmutableMap.of();
         Properties properties = new Properties();
         properties.setProperty("c_time", "en_US");
         properties.setProperty("username", "alice");
-        ConnectionQuerySettings connectionQuerySettings = ConnectionQuerySettings.of(properties);
-        assertThat(connectionQuerySettings.getSettings()).containsExactlyInAnyOrderEntriesOf(expected);
+        ConnectionProperties connectionProperties = ConnectionProperties.of(properties);
+        assertThat(connectionProperties.getStatementProperties().getQuerySettings())
+                .containsExactlyInAnyOrderEntriesOf(expected);
     }
 
     @Test
-    void testGetSettingWithEmptyProperties() {
+    void testRoundtrip() throws DataCloudJDBCException {
+        Properties properties = new Properties();
+        // Cover string setting
+        properties.setProperty("workload", "testWorkload");
+        // Cover prefixed setting
+        properties.setProperty("querySetting.A", "A");
+        properties.setProperty("querySetting.B", "B");
+        // Cover setting that internally is represented as non string (`Duration`)
+        properties.setProperty("queryTimeout", "30");
+        ConnectionProperties connectionProperties = ConnectionProperties.of(properties);
+        Properties roundtripProperties = connectionProperties.toProperties();
+        assertThat(roundtripProperties.entrySet()).containsExactlyInAnyOrderElementsOf(roundtripProperties.entrySet());
```

Comment:
This assertion looks off, I'd expect the `expected` to be from a different object than the `actual`